src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -56,15 +61,47 @@ def __str__(self):
         return "\n".join(map(str, self))
 
 
-def validate_role(role: str, valid_roles: List[str] = None):
+class PromptResult(str):
```

Comment:
add docstring to explain why introduce this class wrapper, and what `original_string`, `escaped_string`, `escaped_mapping` mean.

Reply:
added